apps/ocpp/consumers/base/consumer/connection_flow.py

```
@@ -0,0 +1,87 @@
+    """Service for evaluating if a charge point connection should be admitted."""
+
+    def __init__(
+        self,
+        *,
+        feature_state_resolver: Callable[[Charger | None], AdmissionDecision] | None = None,
+        db_call: Callable[[Callable[[], AdmissionDecision]], Awaitable[AdmissionDecision]] | None = None,
+    ) -> None:
+        self._feature_state_resolver = feature_state_resolver or self._resolve_feature_state
+        self._db_call = db_call
+
+    async def allow_charge_point_connection(self, consumer: "CSMSConsumer", existing_charger: Charger | None) -> bool:
```

Ruff (0.15.0)

[warning] Unused method argument: `consumer`

(ARG002)